Automattic/vip-go-mu-plugins
# Search: Queue class - Rename $object_type to $indexable_slug and throw error if queue_object() has invalid second parameter

## Description
This PR:
- In `class-queue.php`, `$object_type` is renamed to `$indexable_slug` for better clarity 
- In `queue_object()`, a `WP_Error` will be thrown if it is called with an invalid indexable slug. This solves the issue if `$index_version` returns a `WP_Error`, it won't get converted to `0` from `$wpdb->prepare()`
- PHPDoc everything in `class-queue.php`

## Changelog Description

### Plugin Updated: Enterprise Search

Rename `$object_type` to `$indexable_slug` and throw error if `queue_object()` has invalid second parameter

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
1) Shell in and do:
```
$es = \Automattic\VIP\Search\Search::instance();
$es->init();

$es->queue->queue_object( 999, 'foo' );
```
2) Expect to see a `WP_Error` returned:
```
=> object(WP_Error)#4119 (3) {
  ["errors"]=>
  array(1) {
    ["invalid-indexable"]=>
    array(1) {
      [0]=>
      string(32) "Indexable not found for type foo"
    }
  }
  ["error_data"]=>
  array(0) {
  }
  ["additional_data":protected]=>
  array(0) {
  }
}
```
